crypto/serializer/serializer_local.h

```diff
@@ -50,3 +72,58 @@ struct ossl_serializer_ctx_st {
      */
     UI_METHOD *allocated_ui_method;
 };
+
+struct ossl_deserializer_instance_st {
+    OSSL_DESERIALIZER *deser;    /* Never NULL */
+    void *deserctx;              /* Never NULL */
+    const char *input_type;      /* Never NULL */
+    const char *output_type;     /* Never NULL */
+};
+
+DEFINE_STACK_OF(OSSL_DESERIALIZER_INSTANCE)
+
+struct ossl_deserializer_ctx_st {
+    /*
+     * The caller may know the input type of the data they pass.  If not,
+     * this will remain NULL and the deserializing functionality will try
+     * whatever it has in store.
+     */
+    const char *start_input_type;
+
+    /*
+     * Deserializers that are components of any current deserialization path.
+     */
+    STACK_OF(OSSL_DESERIALIZER_INSTANCE) *deser_insts;
+
+    /*
+     * The finalizer of a deserialization, and its caller argument.
+     */
+    OSSL_DESERIALIZER_FINALIZER *finalizer;
+    OSSL_DESERIALIZER_CLEANER *cleaner;
+    void *finalize_arg;
+
+    /* For any function that needs a passphrase reader */
+    const UI_METHOD *ui_method;
+    void *ui_data;
+    /*
+     * if caller used OSSL_SERIALIZER_CTX_set_passphrase_cb(), we need
+     * intermediary storage.
+     */
+    UI_METHOD *allocated_ui_method;
+    /*
+     * Because the same input may pass through more than one deserializer,
+     * we cache any passphrase that's passed to us.  The desrializing
```

Comment:
2 NTS: any passphrase passed to us. The deserializing

Reply:
Done